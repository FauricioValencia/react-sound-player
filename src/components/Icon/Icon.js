import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import ErrorBoundary from 'HOCS/ErrorBoundary'

const Icon = ({ icon, onClick }) => (
  <ErrorBoundary>
    <div className="control" onClick={onClick}>
      <div className="button" id="play-pause-button">
        <FontAwesomeIcon icon={icon} />
      </div>
    </div>
  </ErrorBoundary>
)

Icon.propTypes = {
  icon: PropTypes.object,
  onClick: PropTypes.func,
}

export default Icon
