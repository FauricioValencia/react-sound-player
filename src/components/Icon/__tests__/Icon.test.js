import { shallow } from 'enzyme'

import Icon from '../Icon'

const testProps = { className: 'active', onClick: jest.fn() }

describe('Icon component', () => {
  describe('GIVEN testProps', () => {
    describe('WHEN the component is mounted', () => {
      const wrapper = shallow(<Icon {...testProps} />)
      it('THEN should show correctly', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
