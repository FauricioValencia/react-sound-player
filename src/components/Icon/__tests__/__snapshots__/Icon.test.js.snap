// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component GIVEN testProps WHEN the component is mounted THEN should show correctly 1`] = `ShallowWrapper {}`;
